src/export.rs

```diff
@@ -0,0 +1,235 @@
+        }
+    }
+}
+
+impl From<&PlayedCard> for PokemonState {
+    fn from(card: &PlayedCard) -> Self {
+        PokemonState {
+            card_name: card.card.get_name().to_string(),
+            hp: card.total_hp,
+            remaining_hp: card.remaining_hp,
+            attached_energy: card.attached_energy.iter().map(|e| format!("{:?}", e)).collect(),
+            attached_tool: card.attached_tool.map(|t| format!("{:?}", t)),
+            played_this_turn: card.played_this_turn,
+            poisoned: card.poisoned,
+            paralyzed: card.paralyzed,
+            asleep: card.asleep,
+            stage: match &card.card {
+                crate::types::Card::Pokemon(pokemon) => pokemon.stage,
+                _ => 0, // Shouldn't happen for played cards
+            },
+        }
+    }
+}
+
+impl From<&Action> for ActionData {
+    fn from(action: &Action) -> Self {
+        ActionData {
+            action_type: format!("{:?}", std::mem::discriminant(&action.action)),
+            description: format!("{}", action.action),
+            actor: action.actor,
+        }
+    }
+}
+
+impl Default for TrainingData {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Event handler for collecting training data
+pub struct TrainingDataCollector {
+    pub training_data: TrainingData,
+    current_game: Option<GameData>,
+}
+
+impl TrainingDataCollector {
+    pub fn new() -> Self {
+        Self {
+            training_data: TrainingData::new(),
+            current_game: None,
+        }
+    }
+
+    pub fn save_to_file(&self, path: &str) -> Result<(), Box<dyn std::error::Error>> {
+        self.training_data.save_to_file(path)
+    }
+}
+
+impl Default for TrainingDataCollector {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+use crate::simulation_event_handler::SimulationEventHandler;
+
+impl SimulationEventHandler for TrainingDataCollector {
+    fn on_simulation_start(&mut self) {
+        // Initialize training data
+        self.training_data = TrainingData::new();
+    }
+
+    fn on_game_start(&mut self, game_id: Uuid) {
+        self.current_game = Some(GameData::new(game_id));
+    }
+
+    fn on_action(
+        &mut self,
+        _game_id: Uuid,
+        state_before_action: &State,
+        actor: usize,
+        playable_actions: &[Action],
+        action: &Action,
+        _state_after_action: &State,
+    ) {
+        if let Some(ref mut game) = self.current_game {
+            let decision = Decision {
+                turn: state_before_action.turn_count,
+                acting_player: actor,
+                state: GameState::from(state_before_action),
+                possible_actions: playable_actions.iter().map(ActionData::from).collect(),
+                action_chosen: ActionData::from(action),
+                action_scores: None, // TODO: Add expectiminimax scores if available
+            };
+            game.add_decision(decision);
+        }
+    }
```

Comment:
I am thinking this hook will be the most important one, and that a TrainingDataCollector that just mantains Vectors of the objects given here (game_id, state_before_action, actor, action, ...) and just saves them raw as JSON would be enough for all intents and purposes, no?

If we do this, then this becomes much simpler, which I like! It'll just be a SimulationEventHandler that accumulates clones of everything and saves them in their natural JSON representation (which could be good for non-machine learning purposes as well).